src/core/selinux-setup.c

```diff
@@ -104,3 +107,38 @@ int mac_selinux_setup(bool *loaded_policy) {
 
         return 0;
 }
+
+int mac_selinux_setup_namespaced_policy(const char *policy) {
+
+#if HAVE_SELINUX
+        int r;
+        _cleanup_close_ int fd = -EBADF;
+
+        assert(policy);
+
+        // TODO use selinux_unshare() from libselinux
+        fd = open("/sys/fs/selinux/unshare", O_WRONLY | O_CLOEXEC);
```

Comment:
so, when this fails, there should be a clear log message be shown here, because I understan this requires a very new (unreleased even for now?) kernel. Also, I think we should probably handle this gracefully, i.e. warn but continue?

Reply:
This part here up to `umount()` below should be included in a future version of `libselinux` which should have the new function `selinux_unshare()`.

(The path `/sys/fs/selinux/unshare` exists only with namespace aware kernels, so continuing would not make sense.)